python/lsst/ip/isr/ptcDataset.py

```diff
@@ -196,65 +197,76 @@ def __init__(self, ampNames=[], ptcFitType=None, covMatrixSide=1, **kwargs):
-        """Function to initialize an amp of a PhotonTransferCurveDataset.
-
-        Notes
-        -----
-        The parameters are all documented in `init`.
+    def setAmpValuesPartialDataset(
+            self,
+            ampName,
+            inputExpIdPair=(-1, -1),
+            rawExpTime=np.nan,
+            rawMean=np.nan,
+            rawVar=np.nan,
+            photoCharge=np.nan,
+            expIdMask=False,
+            covariance=None,
+            covSqrtWeights=None,
+            gain=np.nan,
+            noise=np.nan,
+    ):
+        """
+        Set the amp values for a partial PTC Dataset (from cpExtractPtcTask).
+
+        Parameters
+        ----------
+        ampName : `str`
+            Name of the amp to set the values.
+        inputExpIdPair : `tuple` [`int`]
+            Exposure IDs of input pair.
+        rawExpTime : `float`, optional
+            Exposure time for this exposure pair.
+        rawMean : `float`, optional
+            Average of the means of the exposures in this pair.
+        rawVar : `float`, optional
+            Variance of the difference of the exposures in this pair.
+        photoCharge : `float`, optional
+            Integrated photocharge for flat pair for linearity calibration.
+        expIdMask : `bool`, optional
+            Flag setting if this exposure pair should be used (True)
+            or not used (False).
+        covariance : `np.ndarray` or None, optional
+            Measured covariance for this exposure pair.
+        covSqrtWeights : `np.ndarray` or None, optional
+            Measured sqrt of covariance weights in this exposure pair.
+        gain : `float`, optional
+            Estimated gain for this exposure pair.
+        noise : `float`, optional
+            Estimated read noise for this exposure pair.
         """
         nanMatrix = np.full((self.covMatrixSide, self.covMatrixSide), np.nan)
-        if len(covArray) == 0:
-            covArray = [nanMatrix]
-        if len(covArrayModel) == 0:
-            covArrayModel = [nanMatrix]
-        if len(covSqrtWeights) == 0:
-            covSqrtWeights = [nanMatrix]
-        if len(covArrayModelNoB) == 0:
-            covArrayModelNoB = [nanMatrix]
-        if len(aMatrix) == 0:
-            aMatrix = nanMatrix
-        if len(bMatrix) == 0:
-            bMatrix = nanMatrix
-        if len(aMatrixNoB) == 0:
-            aMatrixNoB = nanMatrix
-
-        self.inputExpIdPairs[ampName] = inputExpIdPair
-        self.expIdMask[ampName] = expIdMask
-        self.rawExpTimes[ampName] = rawExpTime
-        self.rawMeans[ampName] = rawMean
-        self.rawVars[ampName] = rawVar
-        self.photoCharge[ampName] = photoCharge
+        if covariance is None:
+            covariance = nanMatrix
+        if covSqrtWeights is None:
+            covSqrtWeights = nanMatrix
+
+        # This first one I'm not sure of the type; list of tuples is okay?
```

Comment:
A list (or NumPy array, if you changed everything to NumPy arrays) of tuples should be fine, I would think. If so, don't forget to delete this comment.

Reply:
Oops, that was a vestigial comment, I'll remove it.